File: src/main/scala/com/rawlabs/das/jira/table/DASJiraTable.scala
/*
 * Copyright 2024 RAW Labs S.A.
 *
 * Use of this software is governed by the Business Source License
 * included in the file licenses/BSL.txt.
 *
 * As of the Change Date specified in that file, in accordance with
 * the Business Source License, use of this software will be governed
 * by the Apache License, Version 2.0, included in the file
 * licenses/APL.txt.
 */

package com.rawlabs.das.jira.table

import com.rawlabs.das.sdk.{DASExecuteResult, DASTable}
import com.rawlabs.protocol.das.{Qual, SortKey}
import com.typesafe.scalalogging.StrictLogging

class DASJiraTable extends DASTable with StrictLogging {

  override def getRelSize(quals: Seq[Qual], columns: Seq[String]): (Int, Int) = ???

  override def execute(
      quals: Seq[Qual],
      columns: Seq[String],
      maybeSortKeys: Option[Seq[SortKey]],
      maybeLimit: Option[Long]
  ): DASExecuteResult = ???
}

File: src/main/java/com/rawlabs/das/jira/tables/DASJiraTable.java
package com.rawlabs.das.jira.tables;

import com.rawlabs.das.sdk.java.DASExecuteResult;
import com.rawlabs.das.sdk.java.DASTable;
import com.rawlabs.das.sdk.java.RowsEstimation;
import com.rawlabs.protocol.das.Qual;
import com.rawlabs.protocol.das.SortKey;
import com.rawlabs.protocol.das.TableDefinition;
import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.Map;

public abstract class DASJiraTable implements DASTable {
  private final Map<String, String> options;
  private final String tableName;
  private final TableDefinition tableDefinition;
  private final RowsEstimation rowsEstimation = new RowsEstimation(100, 100);

  protected DASJiraTable(String tableName, Map<String, String> options) {
    this.options = options;
    this.tableName = tableName;
    this.tableDefinition = buildTableDefinition();
  }

  protected abstract TableDefinition buildTableDefinition();
//  protected abstract TableDefinition buildMappings();


  @Override
  public DASExecuteResult execute(List<Qual> quals, List<String> columns, @Nullable List<SortKey> sortKeys, @Nullable Long limit) {
    return null;
  }

  public String getTableName() {
    return tableName;
  }

  public TableDefinition getTableDefinition() {
    return tableDefinition;
  }

  @Override
  public RowsEstimation getRelSize(List<Qual> quals, List<String> columns) {
    return rowsEstimation;
  }
}
